MAUI/Scheduler/load-on-demand.md

```diff
@@ -0,0 +1,311 @@
+}
+
+private ObservableCollection<SchedulerAppointment> GenerateSchedulerAppointments(List<DateTime> visibleDates)
+{
+    var brush = new ObservableCollection<Brush>
+    {
+        new SolidColorBrush(Color.FromArgb("#FF8B1FA9")),
+        new SolidColorBrush(Color.FromArgb("#FFD20100")),
+        new SolidColorBrush(Color.FromArgb("#FFFC571D")),
+        new SolidColorBrush(Color.FromArgb("#FF36B37B")),
+        new SolidColorBrush(Color.FromArgb("#FF3D4FB5")),
+        new SolidColorBrush(Color.FromArgb("#FFE47C73")),
+        new SolidColorBrush(Color.FromArgb("#FF636363")),
+        new SolidColorBrush(Color.FromArgb("#FF85461E")),
+        new SolidColorBrush(Color.FromArgb("#FF0F8644")),
+        new SolidColorBrush(Color.FromArgb("#FF01A1EF"))
+    };
+
+    var subjectCollection = new ObservableCollection<string>
+    {
+        "Business Meeting",
+        "Conference",
+        "Medical check up",
+        "Performance Check",
+        "Consulting",
+        "Project Status Discussion",
+        "Client Meeting",
+        "General Meeting",
+        "Yoga Therapy",
+        "GoToMeeting",
+        "Plan Execution",
+        "Project Plan"
+    };
+
+    Random ran = new();
+    int daysCount = visibleDates.Count;
+    DateTime visibleStartDate = visibleDates.FirstOrDefault();
+    var appointments = new ObservableCollection<SchedulerAppointment>();
+    for (int i = 0; i < 25; i++)
+    {
+        var startTime = visibleStartDate.AddDays(ran.Next(0, daysCount + 1)).AddHours(ran.Next(9, 16));
+        appointments.Add(new SchedulerAppointment()
+        {
+            StartTime = startTime,
+            EndTime = startTime.AddHours(1),
+            Subject = subjectCollection[ran.Next(0, subjectCollection.Count)],
+            Background = brush[ran.Next(0, brush.Count)]
+        });
+    }
+
+    return appointments;
+}
+
+{% endhighlight %}
+{% endtabs %}
+
+N> The `QueryAppointments` event will be raised if any one of the following action is performed,
+* Once the `ViewChanged` event is raised, the `QueryAppointments` will be raised.
+* If an appointment has been added or removed in the current visible dates, then the `QueryAppointments` event will not be triggered. Because the appointments for that visible date are already loaded.
+* In the scheduler agenda view, the `QueryAppointments` event is used to load more appointments when the new month is loaded on view, whereas the new month appointments are added in scheduler `AppointmentsSource.`
+* Other than agenda view, the scheduler `AppointmentsSource` can be reset for a new visible date range to improve appointment loading performance.
+
+### QueryAppointments command
+
+The Scheduler notifies the `QueryAppointmentsCommand,` when the view of the scheduler or the visible dates are changed. Get a visible dates from the `SchedulerQueryAppointmentsEventArgs.` The default value for this `ICommand` is null. The `SchedulerQueryAppointmentsEventArgs` passed as a command parameter.
```

Comment:
The Scheduler notifies the `QueryAppointmentsCommand,` when the view of the scheduler or the visible dates are changed. Get a list of visible dates from the `SchedulerQueryAppointmentsEventArgs.` The default value for this `ICommand` is null. The `SchedulerQueryAppointmentsEventArgs` is passed as a command parameter.